app/controllers/v2/tailorings_controller.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+module V2
+  # API for Tailorings
+  class TailoringsController < ApplicationController
+    def index
+      render_json tailorings
+    end
+    permission_for_action :index, Rbac::POLICY_READ
+
+    def show
+      render_json tailoring
+    end
+    permission_for_action :show, Rbac::POLICY_READ
+
+    def tailoring_file
+      return unless tailored?
+
+      send_data(
+        xccdf_tailoring_file,
+        filename: xccdf_tailoring_filename,
+        type: Mime[:xml]
+      )
+    end
+    permission_for_action :tailoring_file, Rbac::POLICY_READ
+    permitted_params_for_action :tailoring_file, id: ID_TYPE.required
+
+    private
+
+    def tailorings
+      @tailorings ||= authorize(fetch_collection)
+    end
+
+    def tailoring
+      @tailoring ||= authorize(expand_resource.find(permitted_params[:id]))
+    end
+
+    def tailored?
+      added_rules = tailoring.rules.order(:precedence) - tailoring.profile.rules
+      removed_rules = tailoring.profile.rules - tailoring.rules.order(:precedence)
+      (added_rules + removed_rules).any?
+    end
```

Comment:
This needs a `TODO` comment to check its performance. I believe it is slow 😞 for now it can stay as it is likely copied from the old world, but we need to do it faster.